pom.xml

```diff
@@ -45,12 +45,18 @@
         <webservicesRestVersion>2.29.0</webservicesRestVersion>
         <serializationXstreamVersion>0.2.14</serializationXstreamVersion>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <metdatashairngVersion>1.8.0-SNAPSHOT</metdatashairngVersion>
     </properties>
 
     <dependencyManagement>
         <dependencies>
 
             <!-- Begin OpenMRS modules -->
+            <dependency>
+                <groupId>org.openmrs.module</groupId>
+                <artifactId>metadatasharing-api</artifactId>
+                <version>${metdatashairngVersion}</version>
```

Comment:
Metadata sharing should be marked as `<scope>provided</scope>`, otherwise, you're bundling a version of it in your OMOD, which is likely causing some of the issues. Additionally, in the config.xml file (omod/src/main/resources/config.xml) you'll need to add something like:

```
<require_module version="${metadatashairngVersion}">org.openmrs.module.metadatasharing</require_module>
```

This ensures that the dhisconnector module is only loaded after the metadatasharing module, and so, the class should be available when you need it.

There's a typo in your property name (should be `metadatasharingVersion` or, better, `metadataSharingVersion`)

Finally, are there specific features you need from the `1.8.0-SNAPSHOT`? Otherwise, it might be best to pin this to a slightly older version so that this module can be used without needing to upgrade their version of metadatasharing (maybe something like whatever version was released in RefApp 2.11). Also, just in general, it's best to avoid relying on SNAPSHOT versions if at all possible.


Sorry for all the comments on one line, but it seemed the most relevant place to put them.

Reply:
Oh right, I'll make the changes
Thank you for the feedback @ibacher, it means a lot

Reply:
That fixed the error @ibacher 
Thank you